tensorflow/core/grappler/optimizers/auto_mixed_precision.cc

```diff
@@ -1053,7 +1057,9 @@ class AutoMixedPrecisionImpl {
         target_dtype_((mode_ == AutoMixedPrecisionMode::CUDA ||
                        mode_ == AutoMixedPrecisionMode::CPU)
                           ? DT_HALF
-                          : DT_BFLOAT16) {}
+                          : DT_BFLOAT16),
+        num_gpus_(GetNumGPUs(*cluster)),
+        run_fp16_on_cpu_(run_fp16_on_cpu) {}
```

Comment:
I think adding a new enum value to AutoMixedPrecisionMode, say FP16_CPU, would be better then adding a run_fp16_on_cpu bool.

Reply:
thanks for your review. I am checking /working on this.

Reply:
@reedwm I added a new enum mode AutoMixedPrecisionMode::FP16_CPU.